trieste/acquisition/function.py

```diff
@@ -36,66 +40,59 @@
 BatchAcquisitionFunction handles batches of query points of shape [..., B, D] and returns [..., 1] values.
 """
 
+F = TypeVar("F")
+
 
-class AcquisitionFunctionBuilder(ABC):
-    """ An :class:`AcquisitionFunctionBuilder` builds an acquisition function. """
+class Builder(ABC, Generic[F]):
+    """ Generic builder of things from data and models. """
 
     @abstractmethod
-    def prepare_acquisition_function(
-        self, datasets: Mapping[str, Dataset], models: Mapping[str, ProbabilisticModel]
-    ) -> AcquisitionFunction:
+    def build(self, datasets: Mapping[str, Dataset], models: Mapping[str, ProbabilisticModel]) -> F:
```

Comment:
i remember we changed this from `build` to `prepare_acquisition_function`. This refactor means the `_acquisition_function` bit no longer makes sense, and `prepare` was never really the right word anyway. Can discuss other names if you like